### ISSUE TITLE
Eigene Website für die SWK Magdeburg?

### ISSUE THREAD

**Issue by anrichter** [swkMagdeburg/SWKMagdeburg.de#1]
Moin,

seit einiger Zeit begleitet mich der Gedanke, für die SWK Magdeburg eine eigene dedizierte Webseite aufzusetzen.

**Warum?**
Das hat mehrere Gründe, die ich hier einfach mal auflisten möchte

- Zentrale Anlaufstelle für Interessenten und nur eine URL, die in Meetings angesagt wird, anstatt wie bisher auf Meetup und softwerkskammer.org hinzuweisen
- Verweist dann für Anmeldungen auf Meetup und softwerkskammer.org
- Gesammeltes Wissen z.B. aus unseren Lean Kaffees festhalten
- Anleitungen für Coding Dojos (Aufsetzen von Testprojekten, ...)
- Unseres Sponsoren könnten sich hier präsentieren
- Jobangebote für Softwerker von Firmen aus Magdeburg und Umgebung
- Blog über unsere Meetings - könnte ein super Einblick in unsere Treffen für Neugierige bieten
- ...

Ich denke, da finden sich noch einige weiter Punkte, die wir auf einer separaten Website anbieten könnten.

**Nachteil?**
Natürlich hat so ein Vorhaben auch Nachteile, die ich hier, ohne Anspruch auf Vollständigkeit aufliste:

- Noch eine Seite für die SWK Magdeburg
- Muss aufgesetzt und gepflegt werden
- URL muss bezahlt werdern (ok. 6€ im Jahr sollten machbar sein 😉)

**Technik?**
Meine Gedanken waren bereits bei der Technik (so einfach, wie möglich) und ich war kurz davor die Seite auch schon testweise aufzusetzen (Jekyll) und online zu nehmen (GitHub Pages). Habe mich dann aber doch nochmal zurückgelehnt und nachgedacht 🤔 

Wir sind jede Menge Softwerker und mit Sicherheit gibt es viele Mitglieder, die sich im Web wesentlich besser auskennen, als ich. 
Und last but not least ist geteiltes Leid halbes Leid. Sprich das Aufsetzen und Pflegen der Website muss nicht an einer Person hängen bleiben. 

Deshalb hier mein Aufruf an euch, die Idee ersteinmal zu diskutieren und Ansätze sowie Freiwillige für die Umsetzung zu finden.

let the discussion begin 🤓 

Gruß
Andreas

**Comment by konrad79** [comment]
Ist in jedem Fall eine schöne Idee.
Die Domain habt ihr noch nicht gesichert?

Wäre sicher auch eine schöne Stelle für eine Sammlung der bisherigen Treffen - mit Themen, Downloads und Fotos.

**Comment by anrichter** [comment]
Genau. Eine Übersicht der bisherigen Treffen macht Sinn. Für alle, die später noch einmal Revue passieren lassen wollen, die nicht teilnehmen konnten und nuegierige, die sehen wollen, was bei uns so los ist.

Das meinte ich oben mit dem Blog. Ob das nun in Form eines Blogs oder als Artikelsammlung kann man sich ja noch überlegen.

Die Domain ist noch nicht registriert. Das können wir machen, wenn die Idee Fahrt aufnimmt.

**Comment by darekkay** [comment]
Ich würde an deinem ursprünglichen Plan festhalten und einen Static Site Generator (Markdown) + GitHub Pages verwenden (KISS). Man könnte auch etwas "cooles" ausprobieren (wie Gatsby, Next, ...), aber da wäre die Einstiegshürde für andere höher.

GitHub unterstützt AFAIK nur Jekyll nativ, aber grundsätzlich kann man mit [GitHub Actions](https://github.com/marketplace/actions/deploy-to-github-pages) auch andere Static Site Generators nehmen.

Beim Tooling kann ich gerne unterstützen, beim Design gibt's hier fähigere Leute 😅

**Comment by anrichter** [comment]
Genau. Die Einstiegshürde sollte so gering wie möglich sein. Und eimal aufgesetzt, sollte der Inhalt möglichst über Markdown oder ähnliches gepflegt werden können. Sonst machtse keiner ;)

Bei meiner Recherche bin ich auch über die GitHub Actions aber im Zusammenspiel mit Jekyll gestoßen:
  https://jekyllrb.com/docs/continuous-integration/github-actions/
Denn die native GitHub Pages Unterstützung erlaubt nicht alle Jekyll Plugins. Actions sind da und können imho auch genutzt werden.

Zum Design hab ich mal auf https://jekyllthemes.io gestöbert und bin über Minimal Mistake gestolpert. https://jekyllthemes.io/theme/minimal-mistakes

Ich werde mal die ersten Ziele wie einfache Einstiegshürde, KISS und Editieren mit Markdown festhalten.

**Comment by gaerfield** [comment]
Ich finde Github-Pages ebenfalls logisch. In dem Fall benötigt man absolut gar keine Infrastruktur und erspart sich ein Haufen Arbeit mit der Pflege eines Apache-Servers und den notwendigen Vorsichtsmaßnahmen zur Prävention von irgendwelchen Hacks.
Zusammen mit dem automatischen Build über die Actions ist sogar online editieren möglich ohne git-Kenntnisse. Der Unterschied zu Wordpress ist dann nur noch die notwendige Kenntnis zur Markdown-Syntax.
Das einzige Problem, was ich noch sehe, ist SSL-Terminierung.

Ich weiss nicht, wie wir mit einer eigenen Domain eine ordentliche SSL-Terminierung hinbekommen, wenn die Seite über github gehostet wird.

**Comment by anrichter** [comment]
SSL gibts für GitHub Pages mit Custom Domain out of the box. Laut diesem Blogpost:
  https://github.blog/2018-05-01-github-pages-custom-domains-https/

Sollte also auch kein Problem darstellen.

**Comment by gaerfield** [comment]
Ah ... das läuft wie üblich über einen A-Record ... coole Sache
Und CDN gibts auch dazu ... toppi

**Comment by hepisec** [comment]
Die Domain kann ich kostenfrei beisteuern. Wie sieht es bei Github Pages mit den Zugriffsrechten aus? Änderungen erfolgen dann über Pull-Request und ein Projekt-Maintainer muss die mergen?

**Comment by anrichter** [comment]
Sehr gut. 👍 Domain ist gebongt. 

Genau. Jeder Softwerker, der hier in der Gruppe ist sollte auf das Repository auch schreibrechte haben. Input von Außen kommt dann im Idealfall per Pull Request rein und muss von einem angenommen werden.

**Comment by anrichter** [comment]
Mal abgesehen von der ganzen technischen Kiste und dem Aufsetzen. 
Wer wäre denn bereit sich am Inhalt zu beteiligen?
Sprich Seiten mit Inhalten erstellen, evtl. Blogposts nach unseren Treffen schreiben und was da sonst noch so auftaucht.

**Comment by konrad79** [comment]
Ich finde, stabil Content zu liefern ist das schwerste - viel schwerer als irgendeine technische Grundlage zu bauen. Würde sowas machen, wenn ich öfters dabei wäre.

**Comment by anrichter** [comment]
Na dann. Ein Grund mehr, zu unseren Treffen zu kommen. 😉
